### DISCUSSION TITLE
Deprecate official support to run Renovate on Windows

### DISCUSSION THREAD

**Discussion by viceice** [renovatebot/renovate#32751] [Suggest an Idea]
### Tell us more.

We plan to deprecate official support to run Renovate on Windows.

## Steps planned now
- [x] Disable Windows CI tests (#32754)
- [ ] Add a warning log message, when Renovate is run on Windows with a message pointing to this discussion.
- [ ] Gather comunity feedback
- [ ] Support running Renovate on Windows on best efforts via comunity

## Later steps
-  Make futher decisions.

---

- Please vote with 👍 if you're ok with this deprecation
- Please vote with 👎 if you need Windows support and add a comment why you need it.

**Comment by jmartens** [comment]
Will this also have consequences for the renovate pre-commit hook when doing development on Windows systems? 

I would be really sad if it does as we do a lot of .NET and .NET framework development (on Windows systems) and use renovate as a pre-commit hook. This might mean we would have to integrate it into a build hook to verify renovate configuration of a repository to check if changes to the renovate configuration file are valid. 

This would be a tremendous setback IMHO. Since we target Windows our build hosts are using windows based build hosts.
If I interpret this correctly I cannot even run it (not even conditionally) in our current build pipelines and I would have to filter out commits that modify renovate configuration files, create a seperate azure-pipeline for those commits to run renovate in a build job on a non-windows host.

Apart from that I also debug configuration changes on my windows system using `npx renovate`, will this also be impossible?

**Reply by viceice** [reply]
probably yes. that's why we gather feedback.

as mentioned we can support windows on a best effort base. it's too much overhead and too slow to run CI checks on windows.

**Comment by jmartens** [comment]
I pity you already disabled the windows testing. I stumbled on this message by accident while wanting to raise a new discussion. I have not seen it in my logs as a depreciation message seems to not be in place (according to the checkllist at the top of this issue). Despite that you already seem to have [disabled testing on windows](#32754) even despite a [comment against](https://github.com/renovatebot/renovate/pull/32754#pullrequestreview-2462291850) it. 

IMHO this might mean windows builds might already brake w/o development being aware of it. This might hit users unexpectedly if this will cause failures in the field.

**Reply by rarkins** [reply]
Windows support ultimately slows the project down. Right now, the core team has no use for Windows support (note: I use Windows/WSL though, as do many others). Additionally, Renovate's corporate sponsor (Mend.io, my employer) has no customers needing it either.

As a user, you could either:
- Become a customer of Mend.io, taking the count from 0 to 1, or
- Commit yourself or a developer to maintaining Windows support on Renovate and offloading the burden from the core team

**Reply by jmartens** [reply]
> Windows support ultimately slows the project down. 

Valid and understandable.

> Right now, the core team has no use for Windows support (note: I use Windows/WSL though, as do many others)

WSL is not an option as this would substantially complicate stuff as developers will not have seamless integrations into their GIT clients, like SourceTree or Fork on their native OS (Windows).

> * Become a customer of Mend.io, taking the count from 0 to 1, or

Hahaha, this would hardly make any waves in the pond I guess. I don't have high hopes for being able to tip the balance into keeping Windows support by bumping the number of customers from 0 to 1.

> * Commit yourself or a developer to maintaining Windows support on Renovate and offloading the burden from the core team

If we could miss the scarce resources to dedicate a user to a language that is not in our portfolio this might be a possibility, but I don't think we have the opportunity to do so unfortunately.

I really liked the fact that renovate could be run through a pre-commit hook on any OS, judging from your reaction we might have to say goodbye to renovate pre-commit hook, which I really pity.

**Comment by viceice** [comment]
- https://github.com/apache/datafusion/issues/13726#issuecomment-2549181242

the new window 2025 runner are now in beta, we can test them. the linked issue says it should be ~20 times faster than 2022.

**Reply by jmartens** [reply]
> * [CI: Windows flow takes 1.5h apache/datafusion#13726 (comment)](https://github.com/apache/datafusion/issues/13726#issuecomment-2549181242)
> 
> the new window 2025 runner are now in beta, we can test them. the linked issue says it should be ~20 times faster than 2022.

Is there a way to test them already to see if the gain is worth keeping windows included in CI/CD workflow?

**Reply by viceice** [reply]
- #33237

**Comment by BhaaLseN** [comment]
I currently run Renovate on-premise on Windows, but mostly because it was a machine with a dev setup that I already had. Mostly using .NET Framework and .NET 6/8 projects with it. If I need a Linux going forward, I'll just spin up a new VM for it. Or hopefully get started with Docker soon-ish, maybe on top of GitLab-CI.

What I'm wondering though: _Why_? I'm having a hard time finding a rationale for deprecating the support (other than "it ultimately slows the project down"; which was buried in a discussion comment rather than the initial discussion text). Don't get me wrong, I'm not _against_ it if there is a good reason to do it, but doing things just for the sake of it (and making _me_ do the work of migrating _elsewhere_) leaves a bit of sour aftertaste in my role as user/administrator.

**Reply by rarkins** [reply]
It takes time to keep it working, and that time is better spent on features. The recommended way to run Renovate is via the official docker images.

**Comment by hazendaz** [comment]
Is there a specific code in renovate for windows now or just the builds that were removed. 

We run self hosted and are building out a cli for Windows currently, which is basically ready state.  The only 2 things necessary are using the Windows nodejs to execute it and adjust to git line endings treatment so it matches properly once it hits bitbucket or github properly.  We run off a prebuilt notion that is platform agnostic now.  The compiled bundle is ultimately built on Linux but works on both linux and windows fine this way.  I would not want a depreciation message if the only reason is that it's not tested. We are doing plenty of that and would contribute if the need arises.

Docker simply isn't a viable option for us.  We lost any ability to spin docker from windows, and security issues make it far too complicated to use on Linux as well.  So we figured a well-made setup that we update 1 to 2 times a week to stay current with any changes here works well for our needs and the prebuilt package we produce and curl out of our hosted repository is fast enough to not make docker that appealing.
